Session-4/BubbleSort/Extensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace BubbleSort.Extensions
+{
+    internal static class Extensions
+    {
+        public static void BubbleSort<T>(this T[] array) where T : struct,IComparable
+        {
+            {
+                bool flag = false;
+                if (array.Length == 0)
+                {
+                    Console.WriteLine("Empty array passed");
+                    flag = true;
+                }
+                int num1 = ((IEnumerable<T>)array).Count() - 1;
+                while (!flag)
```

Comment:
Use forloop instead of while loop. When there are known number of iterations - for loop makes sense

Reply:
Actually the number of iterations depends on the elements of the array and so to run the sorting till all the elements are sorted so to check that I used a while loop.

Reply:
Even though it depends on number of elements, you can derive formula for no of iterations right. So for loop makes sense instead of while loop.